frontend/src/hooks/data/contracts.ts

```diff
@@ -0,0 +1,44 @@
+import { useCallback } from "react";
+import { AccessKey } from "../../libraries/wamp/types";
+import { useWampData } from "../use-wamp-data";
+
+interface ContractInfo {
+  codeHash: string;
+  accessKeys: Array<AccessKey>;
+  transactionHash?: string;
+  timestamp?: number;
+}
+
+export const useContractInfo = (accountId: string) =>
+  useWampData<ContractInfo>(
+    "useContractInfo",
+    useCallback(
+      async (wampCall) => {
+        // codeHash does not exist for deleted accounts
+        const account = await wampCall("nearcore-view-account", [accountId]);
+        const codeHash = account["code_hash"];
+        // TODO: find out why we have magic number here
```

Comment:
If account exists it always has codeHash. That is due to the fact that each account on NEAR can have a contract deployed to it (one or none contract can be there), and when there is some contract deployed, it will have 32 bytes of sha256 hash, otherwise, if no contract has ever been deployed to an account, it will be 32 zero bytes which are encoded as 32 "1" in base58 encoding.

Reply:
Thanks!
The constant probably should be called `NO_CONTRACT_FLAG` or something like that.
I'll leave a link to your comment for now and will fix it later.